vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -0,0 +1,466 @@
+    until: new_image_cp_nodes.stdout|int == 3
+    failed_when: new_image_cp_nodes.stdout|int != 3
+
+  - name: Wait for old etcd instance to leave the new etcd-cluster
+    pause:
+      minutes: 2
+
+  - name: Untaint all CP nodes after upgrade of controlplane nodes
+    shell: |
+        kubectl taint nodes --all node-role.kubernetes.io/master-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    ignore_errors: yes
+
+  - name: Verify that the old controlplane node has left the cluster
+    shell: |
+            kubectl get bmh -n {{NAMESPACE}} | grep -i provisioned | grep -c 'test1-controlplane-'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    retries: 200
+    delay: 20
+    register: upgraded_cp_nodes_count
+    until: upgraded_cp_nodes_count.stdout|int == 0
+    failed_when: upgraded_cp_nodes_count.stdout|int != 0
+
+  - name: Scale worker up to 1
+    shell: |
+        kubectl scale machinedeployment "{{ CLUSTER_NAME }}"  -n "{{ NAMESPACE }}" --replicas=1
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Wait until worker is scaled up and no bmh is in Ready state
+    shell: kubectl get node | awk 'NR>1'| grep -cv master
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    retries: 200
+    delay: 20
+    register: worker_nodes
+    until: worker_nodes.stdout|int == 1
+    failed_when: worker_nodes.stdout|int == 0
+
+  - name: Label worker for scheduling purpose
+    shell: |
+      WORKER_NAME=$(kubectl get nodes -n {{NAMESPACE}} | awk 'NR>1'| grep -v master | awk '{print $1}')
+      kubectl label node "${WORKER_NAME}" type=worker
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Copy workload manifest to /tmp
+    copy:
+      src: workload.yaml
+      dest: /tmp/workload.yaml
+
+  - name: Deploy workload with nodeAffinity 
+    shell: |
+        kubectl apply -f /tmp/workload.yaml
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - pause:
+      minutes: 5
+
+  - name: Show workload deployment status on worker node
+    shell: |
+            kubectl get pods | grep 'workload-1-deployment'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Verify workload deployment 
+    shell: |
+            kubectl get deployments workload-1-deployment -o json | jq '.status.readyReplicas'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    retries: 200
+    delay: 20
+    register: running_workload_pods
+    until: running_workload_pods.stdout|int == 10
+    failed_when: running_workload_pods.stdout|int != 10
+
+  - name: Update boot-disk and kubernetes versions of worker node
+    shell: |
+            kubectl get machinedeployment -n {{NAMESPACE}} {{ CLUSTER_NAME }} -o json |
+            jq '.spec.template.spec.infrastructureRef.name="test1-new-workers-image" |
```

Comment:
test1 should be "{{ CLUSTER_NAME }}" ?

Reply:
No, this is a self declared string. Nothing to do with the cluster.

Reply:
I meant test1 prefix is actually coming from "{{ CLUSTER_NAME }}", if you are hardcoding it then its a different issue.

Reply:
replaced